packages/daemon/src/utils/wallet.ts

```diff
@@ -294,11 +296,11 @@ export const prepareInputs = (inputs: EventTxInput[], tokens: string[]): TxInput
       // @ts-ignore
       script: utxo.script,
       token,
-      decoded: {
+      decoded: output.decoded ? {
         type: output.decoded.type,
         address: output.decoded.address,
         timelock: output.decoded.timelock,
-      },
```

Comment:
Is this decoded from the fullnode or local?
A data output has a valid `decoded` with `type` and `data` but it does not have an `address`.

Reply:
The fullnode sends it already decoded, exactly with this structure (including the `address`). It comes from the event queue